src/main/java/filter/expression/Qualifier.java

```diff
@@ -460,6 +460,33 @@ public static boolean hasUpdatedQualifier(FilterExpression expr) {
                     }
                 };
                 break;
+            case "milestone":
+                comparator = (a, b) -> {
+                    Optional<TurboMilestone> aMilestone = model.getMilestoneOfIssue(a);
+                    Optional<TurboMilestone> bMilestone = model.getMilestoneOfIssue(b);
+
+                    if (!aMilestone.isPresent() && !bMilestone.isPresent()) {
+                        return 0;
+                    } else if (!aMilestone.isPresent()) {
+                        return 1;
+                    } else if (!bMilestone.isPresent()) {
+                        return -1;
+                    } else {
+                        Optional<LocalDate> aDueDate = aMilestone.get().getDueDate();
+                        Optional<LocalDate> bDueDate = bMilestone.get().getDueDate();
+
+                        if (!aDueDate.isPresent() && !bDueDate.isPresent()) {
+                            return 0;
+                        } else if (!aDueDate.isPresent()) {
+                            return 1;
+                        } else if (!bDueDate.isPresent()) {
+                            return -1;
+                        } else {
+                            return -aDueDate.get().compareTo(bDueDate.get());
+                        }
+                    }
+                };
+                break;
```

Comment:
The last outer else block can be flattened with @HansNewbie's method in #1050 so either of us can do it depending on which one gets merged first.